src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -298,18 +298,6 @@ private function registerReceivers(ContainerBuilder $container, array $busIds)
         }
 
         $container->getDefinition('messenger.receiver_locator')->replaceArgument(0, $receiverMapping);
-
-        $failedCommandIds = [
-            'console.command.messenger_failed_messages_retry',
-            'console.command.messenger_failed_messages_show',
-            'console.command.messenger_failed_messages_remove',
-        ];
-        foreach ($failedCommandIds as $failedCommandId) {
-            if ($container->hasDefinition($failedCommandId)) {
-                $definition = $container->getDefinition($failedCommandId);
-                $definition->replaceArgument(1, $receiverMapping[$definition->getArgument(0)]);
-            }
-        }
```

Comment:
Hmm, wait a second. This is subtle. But currently, in `FrameworkExtension`, you are populating arg 0 with `$senderReferences[$config['failure_transport']]`. But before, we were populating them with *receiver* references. Even for me, this makes my head spin a bit :P. You could technically have a failure transport "receiver" service but no "sender" service... though this is (at the very least) an edge case. Usually a sender & receiver are a "pair" because you've defined them as a transport. But this doesn't need to be the case.

The situation would be this: I create a `ReceiverInterface` service and tag it with `messenger.receiver` and `alias` `my_failure_receiver`. But, I do not actually register this as a transport. Then, at the command line, I use `messenger:failure:show --failure-transport=my_failure_receiver`. That is technically legal. And it's even possible (though unlikely) to set up this kind of situation if you're using AMQP (e.g. you actually set up a true `failure_transport` called `failure1`, but then with custom binding & routing rules, those messages ultimately end up in a queue connected with the custom `my_failure_receiver` receiver.

So.... that's why this code was originally here and I think it probably still needs to be. I don't think that's a problem, other than some of the new code in `FrameworkExtension` should actually live here.